### DISCUSSION TITLE
Is there a peewee `AutoIncrementField` equivalent?

### DISCUSSION THREAD

**Discussion by loynoir** [sqlalchemy/sqlalchemy#7632] [Ideas]
### Background
I'm SQL newbee on day 2. Sorry first, if I miss some doc. 

I heard `sqlalchemy` kind of complex. But I saw `sqlalchemy` support async, so wanna have a try.

### Question
Is there a `peewee` `AutoIncrementField` equivalent?

### Explain
In peewee, 
- `AutoIncrementField` = max ever value + 1
- `AutoField` = max existing value + 1

https://docs.peewee-orm.com/en/latest/peewee/api.html?highlight=dangerous#AutoField

https://docs.peewee-orm.com/en/latest/peewee/sqlite_ext.html#AutoIncrementField

**Comment by loynoir** [comment]
### My not successful trying

<details>
  <summary></summary>
  <p>
 
```python
from sqlalchemy import create_engine
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import sessionmaker
from sqlalchemy import Column, Integer, String

engine = create_engine("sqlite://")
Base = declarative_base()
Session = sessionmaker(bind=engine)
session = Session()


class Message(Base):
    __tablename__ = "messages"

    id = Column(Integer, primary_key=True, autoincrement=True)
    message = Column(String)


Base.metadata.create_all(engine)
message1 = Message(message="Hello World! 1")
message2 = Message(message="Hello World! 2")
message3 = Message(message="Hello World! 3")

# simulate
session.add(message1)
session.commit()

session.add(message2)
session.commit()
session.delete(message2)
session.commit()

session.add(message3)
session.commit()

query = session.query(Message)
results = query.all()

assert results[1].message == 'Hello World! 3'

# Error: id is 2
assert results[1].id == 3
```
  </p>
</details>

### Expected
```python
>>> assert results[1].message == 'Hello World! 3'
>>> results[1].id
3
```

### Actual
```python
>>> assert results[1].message == 'Hello World! 3'
>>> results[1].id
2
```

**Comment by loynoir** [comment]
Now it works!

```python
class Message(Base):
    __tablename__ = "messages"
    __table_args__ = {'sqlite_autoincrement': True}
    
    id = Column(Integer, primary_key=True, autoincrement=True)
    message = Column(String)
```

Reference

https://github.com/sqlalchemy/sqlalchemy/discussions/7636

**Reply by CaselIT** [reply]
Note that sqlite autoincrement is not needed, since in sqlite a integer primary key automatically autoincrements.

The sqlite docs advise against using it https://sqlite.org/autoinc.html

**Reply by loynoir** [reply]
> Note that sqlite autoincrement is not needed, since in sqlite a integer primary key automatically autoincrements.
> 
> The sqlite docs advise against using it https://sqlite.org/autoinc.html

I read it, but still many thanks for the link. The use case here want the id field to be historically uniq
